Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+    }
+
+    private int scaleDpToPixels(double value, Resources res) {
+        final float scale = res.getDisplayMetrics().density;
+        return (int) Math.round(value * scale);
+    }
+
+    private Drone getDroneApi() { return mDpApp.getDrone(); }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+
+        // Initialize Baidu Map library
+        com.baidu.mapapi.SDKInitializer.initialize(getActivity().getApplicationContext());
+        mDpApp = (DroidPlannerApp) activity.getApplication();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup viewGroup, Bundle bundle) {
+        setHasOptionsMenu(true);
+        final Context context = getActivity().getApplicationContext();
+        final View view = super.onCreateView(inflater, viewGroup, bundle);
+
+        mAppPrefs = DroidPlannerPrefs.getInstance(context);
+
+        final Bundle args = getArguments();
+        if (args != null) {
+            maxFlightPathSize = args.getInt(EXTRA_MAX_FLIGHT_PATH_SIZE);
+        }
+
+        return view;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        LocalBroadcastManager.getInstance(getActivity().getApplicationContext())
+                .registerReceiver(eventReceiver, mEventFilter);
+
+        setupMap();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+
+        LocalBroadcastManager.getInstance(getActivity().getApplicationContext())
+                .unregisterReceiver(eventReceiver);
+
+		mBDLocClient.stop();                       // close baidu location service
+		getBaiduMap().setMyLocationEnabled(false); // disable location layer
+    }
+
+    @Override
+    public void clearFlightPath() {
+        if (mFlightPath != null) {
+            List<LatLng> oldFlightPath = mFlightPath.getPoints();
+            oldFlightPath.clear();
+            mFlightPath.setPoints(oldFlightPath);
+        }
+    }
+
+    @Override
+    public void downloadMapTiles(MapDownloader mapDownloader, VisibleMapArea mapRegion, int minimumZ, int maximumZ) {
+    }
+
+    @Override
+    public LatLong getMapCenter() {
+        return LatLngToCoord(getBaiduMap().getMapStatus().target);
+    }
+
+    @Override
+    public float getMapZoomLevel() {
+        return getBaiduMap().getMapStatus().zoom;
+    }
+
+    @Override
+    public float getMaxZoomLevel() {
+        return getBaiduMap().getMaxZoomLevel();
+    }
+
+    @Override
+    public float getMinZoomLevel() { return getBaiduMap().getMinZoomLevel(); }
+
+    @Override
+    public void selectAutoPanMode(AutoPanMode target) {
+        final AutoPanMode currentMode = mPanMode.get();
+        if (currentMode == target) return;
+
+        setAutoPanMode(currentMode, target);
+    }
+
+    private void setAutoPanMode(AutoPanMode current, AutoPanMode update) {
+        if (mPanMode.compareAndSet(current, update)) {
```

Comment:
@droneboost here you can just call `mPanMode.compareAndSet(current, update)` to accomplish the intended functionality.
You especially don't want to unregister the event receiver since it's filtering over events besides the drone gps updates.